### PR TITLE
cluster v2 Check module

### PR DESCRIPTION
**Pull request by shy-Xu**
### Describe what this PR does / why we need it
this PR is for solving issue#872

### CODE REVIEW

**Comment by shy-Xu** [comment]
checker module has a function in file host_checker.go,this func calls another func included in package ssh.But package ssh has not upgraded to cluster v2,so there is a error exist.

**Comment by fanux** [comment]
Not modify the v1 code, add v2 into pkg/checker, we delete v1 when all test access.

File: pkg/checker/host_checker.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	v2 "github.com/alibaba/sealer/types/api/v2"
+
+	"github.com/alibaba/sealer/utils/ssh"
+)
+
+type HostChecker struct {
+}
+
+func (a HostChecker) Check(cluster *v2.Cluster, phase string) error {
+	ssh := ssh.NewSSHClient(&cluster.Spec.SSH)
```
Comment (fanux) [inline pkg/checker/host_checker.go:32]: Each node may has it own ssh config, checkout this: https://github.com/alibaba/sealer/blob/main/pkg/runtime/kubeadm_runtime.go#L94

1. move the function to ssh module,  GetHostSSHClientByIP(hostIP string, *v2.Cluster)
2. The function has bug, it don't merge the ssh config when host ssh config not exist.

Comment (fanux) [reply pkg/checker/host_checker.go:32]: I will fix those issue.

File: pkg/checker/host_checker.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+func NewHostChecker() Interface {
+	return &HostChecker{}
+}
+
+func checkHostnameUnique(cluster *v2.Cluster, ipList []string) error {
+	hostnameList := map[string]bool{}
+	for _, ip := range ipList {
+		s, err := ssh.GetHostSSHClient(ip, cluster)
+		if err != nil {
+			return err
```
Comment (fanux) [inline pkg/checker/host_checker.go:51]: wrap error.

**Review by fanux — APPROVED** [review]

**Pull request merged**